Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -0,0 +1,120 @@
+#include "waypointManager.hpp"
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+class commsWithAttitude : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        commsWithAttitude() {}
+        commsWithAttitude(const commsWithAttitude& other);
+        commsWithAttitude& operator =(const commsWithAttitude& other);
+};
+
+class getFromTelemetry : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        getFromTelemetry() {}
+        getFromTelemetry(const getFromTelemetry& other);
+        getFromTelemetry& operator =(const getFromTelemetry& other);
+};
+
+class getSensorData : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static AltimeterData_t *GetAltimeterOutput(void) {return &_altimeterdata;}
+        static GpsData_t *GetGPSOutput(void) {return &_gpsdata;}
+    private:
+        getSensorData() {}
+        getSensorData(const getSensorData& other);
+        getSensorData& operator =(const getSensorData& other);
+        static AltimeterData_t _altimeterdata; 
+        static GpsData_t _gpsdata; 
+};
+
+class sensorFusion : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        sensorFusion() {}
+        sensorFusion(const sensorFusion& other);
+        sensorFusion& operator =(const sensorFusion& other);
+};
+
+class cruisingState : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static _PathData *GetWaypointData(void) {return &_waypointdata;}
+        static _WaypointManager_Data_Out *GetOutputData(void) {return &_outputdata;}
+    private:
+        cruisingState() {}
+        cruisingState(const cruisingState& other);
+        cruisingState& operator =(const cruisingState& other);
+        //WaypointManager(0,0) {} initializing instance of WaypointManager class
```

Comment:
We can initialize the waypoint manager class similar to how sensor class instances are initialized in attitude manager!

Since the waypoint manager needs reference coordinates when being initialized, we could create a flag that is sent by telemetry when they send in the original flight path. This flag, when turned on, means that we need to initialize the waypointManager object for this state. Along with this flag will be two coordinates which will be our reference coordinates. 

Then, using an if statement in the execute() method, if the flag is on, we initialize the instance of the waypoint manager!